docs/core/v3.0.rst

```diff
@@ -1095,6 +1091,18 @@ The following names are optional:
     specification. When the ``storage_transformers`` name is absent no storage
     transformer is used, same for an empty list.
 
+``dimension_names``
+^^^^^^^^^^^^^^^^^^^
+
+    Specifies dimension names, e.g. ``["x", "y", "z"]``.  If specified, must be
+    an array of strings with the same length as ``shape``.  An unnamed dimension
+    is indicated by an empty string, ``""``.  Within a single zarr array, all
```

Comment:
Why not use `null` instead for unnamed dimensions?

Reply:
I would prefer enforcing this at the type level, e.g., "dimension names are any valid string."

Otherwise, I imagine _somebody_ is going to try to set dimension names to an empty string in their application code and wonder why they don't round-trip to Zarr properly.

Reply:
The current proposal will allow any string to round trip ---- the uniqueness constraint just applies to non-empty strings.  Are you proposing that empty strings should be supported and be treated exactly the same as  non-empty strings --- and that unnamed dimensions should be identified separately using null?

Reply:
> Are you proposing that empty strings should be supported and be treated exactly the same as non-empty strings --- and that unnamed dimensions should be identified separately using null?

Yes. Otherwise application code will not be able to make the distinction (if desired) between "unnamed" and "named with an empty string."